machine_learning/benchmarks/base_regressor_benchmark.py

```diff
@@ -0,0 +1,135 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Base for Regressor benchmarks."""
+
+from abc import ABC
+from typing import Optional
+
+from qiskit import Aer, QuantumCircuit
+from qiskit.algorithms.optimizers import Optimizer
+from qiskit.circuit import Parameter
+from qiskit.circuit.library import ZZFeatureMap, RealAmplitudes
+from qiskit.utils import QuantumInstance, algorithm_globals
+from qiskit_machine_learning.algorithms import NeuralNetworkRegressor
+from qiskit_machine_learning.neural_networks import TwoLayerQNN
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import MinMaxScaler
+
+from .datasets import (
+    DATASET_SYNTHETIC_REGRESSION_FEATURES,
+    DATASET_SYNTHETIC_REGRESSION_LABELS,
+    load_ccpp,
+)
+
+DATASET_SYNTHETIC_REGRESSION = "dataset_synthetic_regression"
+DATASET_CCPP_REGRESSION = "dataset_ccpp"
+
+
+class BaseRegressorBenchmark(ABC):
+    """Base for Regressor benchmarks."""
+
+    def __init__(self):
+
+        quantum_instance_statevector = QuantumInstance(
+            Aer.get_backend("statevector_simulator"),
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
+        quantum_instance_qasm = QuantumInstance(
+            Aer.get_backend("qasm_simulator"),
+            shots=1024,
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
+
+        self.backends = {
+            "statevector_simulator": quantum_instance_statevector,
+            "qasm_simulator": quantum_instance_qasm,
+        }
+
+        # prepare synthetic dataset
+        (
+            synth_train_features,
+            synth_test_features,
+            synth_train_labels,
+            synth_test_labels,
+        ) = train_test_split(
+            DATASET_SYNTHETIC_REGRESSION_FEATURES,
+            DATASET_SYNTHETIC_REGRESSION_LABELS,
+            test_size=5,
+            shuffle=False,
+        )
+
+        # prepare CCPP dataset, we can afford only a tiny subset of the dataset for training
+        ccpp_features, ccpp_labels = load_ccpp()
+        ccpp_features = ccpp_features[:25]
+        ccpp_labels = ccpp_labels[:25]
+
+        scaler = MinMaxScaler((-1, 1))
+        ccpp_features = scaler.fit_transform(ccpp_features)
+        ccpp_labels = scaler.fit_transform(ccpp_labels.reshape(-1, 1))
+
+        (
+            ccpp_train_features,
+            ccpp_test_features,
+            ccpp_train_labels,
+            ccpp_test_labels,
+        ) = train_test_split(ccpp_features, ccpp_labels, test_size=5, shuffle=False)
```

Comment:
I would consider making it a separate method, e.g. ```_prepare_ccpp_dataset(...)```, for improved readability.

Reply:
Extracted a method that loads the dataset, pick a subset and rescale it. I keep `train_test_split` in the constructor as well as for the synthetic dataset.